crates/solver/src/solver/http_solver.rs

```diff
@@ -367,6 +370,20 @@ impl Solver for HttpSolver {
         if orders.is_empty() {
             return Ok(Vec::new());
         };
+        let prices = external_prices.clone().into_http_solver_prices();
+        orders.extend(
+            liquidity
+                .iter()
+                .filter_map(|liquidity| match liquidity {
+                    Liquidity::LimitOrder(order) => Some(order.clone()),
+                    _ => None,
+                })
+                .filter(|order| {
+                    prices.get(&order.sell_token).is_some()
+                        && prices.get(&order.buy_token).is_some()
```

Comment:
I'm not entirely sure this is needed because of:
```rust
        // We are guaranteed to have price estimates for all tokens that are relevant to the
        // objective value by the driver. It is possible that we have AMM pools that contain tokens
        // that are not any order's tokens. We used to fetch these extra prices but it would often
        // slow down the solver and the solver can estimate them on its own.
        let price_estimates = external_prices.into_http_solver_prices();
```
But when I passed all orders I always got [settlement simulation failures](https://dashboard.tenderly.co/gp-v2/staging/simulator/72e65804-2bf8-4b5b-ad52-dc6b637a5b6f). I think the error is always caused by some transfer exceeding the balance of our settlement contract. I'm not sure how filtering out the orders without prices affects that but it seems to be related.
Any ideas on that would be greatly appreciated.

Reply:
Hmm, looking at the settlement, something looks really suspicious. For a single trade, the solver decided to use 7 0x orders and a Balancer interaction.
1. `0x2b59...` -> `0xa0b8...`
2. `0xa0b8...` -> `0x2b59...`
3. `0x3819...` -> `0xa0b8...`
4. `0x2b59...` -> `0xa0b8...`
5. `0x2b59...` -> `0x3819...`
6. `0x2b59...` -> `0xa0b8...`
7. `0xa0b8...` -> `0x3819...`

And finally, the Balancer V2 swap was:
7. `0xa0b8...` -> `0xdac1...`


Which... looks wrong for matching an order trading `0xa0b8...` to `0xdac1...`

> I'm not sure how filtering out the orders without prices affects that but it seems to be related.

I'm guessing that fewer 0x orders were considered to the problem didn't manifest itself.

Reply:
Also worth noting is that **execution order matters** for 0x orders (and not for GP orders). cc @marcovc

Reply:
It's not obvious to me why execution order would matter for 0x orders. If their price is constant it shouldn't matter in which order they get executed because they can't influence each others prices, no?

Reply:
Its because they are executed immediately, so the settlement contract needs sufficient balance at execution time, and not at the end of the settlement.

Reply:
Thanks for letting me know about this. I think we then need to add some flag to these orders to differentiate them from other liquidity orders where order of execution does not matter, and then do something different with them in the solver if this flag is set.

Reply:
This is actually quite interesting. I ran the numbers and computed how much of each token was leaving the settlement contract and how much was being added to see why all these 0x orders were being traded, and it looks like the optimization solver found some arbitrage opportunity with the orders:
```
┌───────┬──────────────────────────────────────────────┬─────────────┐
│ (idx) │ tokenAddress                                 │ value       │
├───────┼──────────────────────────────────────────────┼─────────────┤
│     0 │ "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48" │ "32130726"  │
│     1 │ "0x2b591e99afE9f32eAA6214f7B7629768c40Eeb39" │ "-51"       │
│     2 │ "0x3819f64f282bf135d62168C1e513280dAF905e06" │ "110326224" │
└───────┴──────────────────────────────────────────────┴─────────────┘
```

Specifically, around `32 USDC` - which isn't a lot. I'd be interested in understanding why it chose to do this (as I imagine the gas we spend is WAY more than 32$). I imagine this could indicate that there is an issue with our fee/cost model to the orders that are being passed in.

@MartinquaXD, maybe you can collect the instance JSON for this kind of settlement to give to @marcovc in order to investigate?

All in all though - it looks like the solver is doing what it is supposed to (modulo including a bunch of orders for marginal amount of arbitrage, although that may be an issue with how the limit orders are being passed in).